geode-core/src/integrationTest/java/org/apache/geode/cache/client/internal/AutoConnectionSourceImplJUnitTest.java

```diff
@@ -96,7 +96,7 @@
   private PoolStats poolStats;
 
   @Rule
-  public final RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
```

Comment:
Is there some reason that this can no longer be final?  Looks like we still instantiate and assign in-line.

Reply:
I'm just following JUnit convention. You'll find that none of the books, articles or examples mark the rules as `final`. It's ok for it to be `final` but I strive for my code to look as much like the books, articles and examples so as not to slow down someone who is reading the code or is looking at the code as an example. Put final rules in one test and you'll see developers start to spread that to other tests. Again, it's not that it's not ok, it's just that if we want to follow the best practices and coding conventions that are put forth in books, articles and other examples, then we should not put `final` on JUnit rules.